daft/catalog/__iceberg.py

```diff
@@ -79,7 +80,8 @@ def _try_from(obj: object) -> IcebergTable | None:
             return IcebergTable(obj)
         return None
 
-    def read(self) -> DataFrame:
-        import daft
+    def read(self, **options) -> DataFrame:
```

Comment:
Why not just `=snapshot_id` as well?

Either that, or we can have a `IcebergTableReadOptions` where we can have `IcebergTableReadOptions.from_dict(...)` throw an appropriate ValueError if any weird options are passed in by the user? That seems better than silently ignoring any odd stuff that users might be passing into the API.

Reply:
> That seems better than silently ignoring any odd stuff that users might be passing into the API.

Agreed, let me patch up these options